### ISSUE TITLE
Module ics not found

### ISSUE THREAD

**Issue by alansill** [koesterlab/calfreeslots#2]
I tried this out in a conda environment (conda create -n calfreeslots python=3) and encountered a "module not found" for ics. A detailed installation log is below in case it is helpful, with the link to the calendar obscured.

```
alansilli7mini:~ alansill$ conda activate calfreeslots
(calfreeslots) alansilli7mini:~ alansill$ which pip
/usr/local/miniconda3/envs/calfreeslots/bin/pip
(calfreeslots) alansilli7mini:~ alansill$ python --version
Python 3.10.4
(calfreeslots) alansilli7mini:~ alansill$ pip install calfreeslots
Collecting calfreeslots
  Downloading calfreeslots-1.0.0-py3-none-any.whl (3.2 kB)
Collecting requests<3.0.0,>=2.28.1
  Downloading requests-2.28.1-py3-none-any.whl (62 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 62.8/62.8 kB 1.8 MB/s eta 0:00:00
Collecting icalendar<5.0.0,>=4.1.0
  Downloading icalendar-4.1.0-py2.py3-none-any.whl (74 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 74.8/74.8 kB 5.9 MB/s eta 0:00:00
Collecting arrow<2.0.0,>=1.2.3
  Downloading arrow-1.2.3-py3-none-any.whl (66 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 66.4/66.4 kB 6.5 MB/s eta 0:00:00
Collecting recurring-ical-events<2.0.0,>=1.0.2-beta.0
  Downloading recurring_ical_events-1.0.2b0-py3-none-any.whl (14 kB)
Collecting python-dateutil>=2.7.0
  Downloading python_dateutil-2.8.2-py2.py3-none-any.whl (247 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 247.7/247.7 kB 7.0 MB/s eta 0:00:00
Collecting pytz
  Downloading pytz-2022.2.1-py2.py3-none-any.whl (500 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 500.6/500.6 kB 13.4 MB/s eta 0:00:00
Collecting x-wr-timezone>=0.0.5
  Downloading x_wr_timezone-0.0.5-py3-none-any.whl (9.8 kB)
Collecting urllib3<1.27,>=1.21.1
  Downloading urllib3-1.26.12-py2.py3-none-any.whl (140 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 140.4/140.4 kB 11.3 MB/s eta 0:00:00
Collecting charset-normalizer<3,>=2
  Downloading charset_normalizer-2.1.1-py3-none-any.whl (39 kB)
Collecting idna<4,>=2.5
  Downloading idna-3.4-py3-none-any.whl (61 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 61.5/61.5 kB 6.9 MB/s eta 0:00:00
Requirement already satisfied: certifi>=2017.4.17 in /usr/local/miniconda3/envs/calfreeslots/lib/python3.10/site-packages (from requests<3.0.0,>=2.28.1->calfreeslots) (2022.9.14)
Collecting six>=1.5
  Downloading six-1.16.0-py2.py3-none-any.whl (11 kB)
Installing collected packages: pytz, urllib3, six, idna, charset-normalizer, requests, python-dateutil, icalendar, arrow, x-wr-timezone, recurring-ical-events, calfreeslots
Successfully installed arrow-1.2.3 calfreeslots-1.0.0 charset-normalizer-2.1.1 icalendar-4.1.0 idna-3.4 python-dateutil-2.8.2 pytz-2022.2.1 recurring-ical-events-1.0.2b0 requests-2.28.1 six-1.16.0 urllib3-1.26.12 x-wr-timezone-0.0.5
(calfreeslots) alansilli7mini:~ alansill$ env | grep -i cal
CONDA_PROMPT_MODIFIER=(calfreeslots) 
CONDA_EXE=/usr/local/miniconda3/bin/conda
PATH=/usr/local/miniconda3/envs/calfreeslots/bin:/Users/alansill/perl5/bin:/usr/local/miniconda3/condabin:/Library/Frameworks/Python.framework/Versions/3.6/bin:/Library/Frameworks/Python.framework/Versions/3.6/bin:/Library/Frameworks/Python.framework/Versions/3.5/bin:/Library/Frameworks/Python.framework/Versions/3.4/bin:/usr/local/sbin:/usr/local/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/TeX/texbin:/usr/local/go/bin:/opt/X11/bin:/Applications/Server.app/Contents/ServerRoot/usr/bin:/Applications/Server.app/Contents/ServerRoot/usr/sbin:/Library/Apple/usr/bin:/usr/local/git/bin:/Users/alansill/.rvm/bin
CONDA_PREFIX=/usr/local/miniconda3/envs/calfreeslots
PERL_LOCAL_LIB_ROOT=/Users/alansill/perl5
CONDA_PYTHON_EXE=/usr/local/miniconda3/bin/python
CONDA_DEFAULT_ENV=calfreeslots

# First try using ics file as the calendar address. See https://support.pushpay.com/s/article/How-do-I-get-an-iCal-link-from-Office-365 for details

(calfreeslots) alansilli7mini:~ alansill$ export CALFREESLOTS_ICAL_URLS="https://outlook.office365.com/owa/calendar/[redacted]/calendar.ics"
(calfreeslots) alansilli7mini:~ alansill$ export CALFREESLOTS_MORNING_START="8:00"
(calfreeslots) alansilli7mini:~ alansill$ export CALFREESLOTS_EVENING_END="17:00" # End time in the evening.
(calfreeslots) alansilli7mini:~ alansill$ export CALFREESLOTS_SHRINK_SLOTS_MIN="5" # Minutes to shring free slots, in order to generate some free time between events.
(calfreeslots) alansilli7mini:~ alansill$ export CALFREESLOTS_MIN_SLOT_LEN="0:30"
(calfreeslots) alansilli7mini:~ alansill$ calfreeslots
Traceback (most recent call last):
  File "/usr/local/miniconda3/envs/calfreeslots/bin/calfreeslots", line 5, in <module>
    from calfreeslots import main
  File "/usr/local/miniconda3/envs/calfreeslots/lib/python3.10/site-packages/calfreeslots/__init__.py", line 6, in <module>
    from ics import Calendar
ModuleNotFoundError: No module named 'ics'

# Second try using html file as the calendar address. 

(calfreeslots) alansilli7mini:~ alansill$ export CALFREESLOTS_ICAL_URLS="https://outlook.office365.com/owa/calendar/[redacted]/calendar.html"
(calfreeslots) alansilli7mini:~ alansill$ calfreeslots
Traceback (most recent call last):
  File "/usr/local/miniconda3/envs/calfreeslots/bin/calfreeslots", line 5, in <module>
    from calfreeslots import main
  File "/usr/local/miniconda3/envs/calfreeslots/lib/python3.10/site-packages/calfreeslots/__init__.py", line 6, in <module>
    from ics import Calendar
ModuleNotFoundError: No module named 'ics'

```

**Comment by alansill** [comment]
The ics module not found issue can be resolved by "pip install ics". I encountered a further error parsing the Microsoft ics file from the link mentioned above, but that is a separate issue.

**Comment by johanneskoester** [comment]
The ics import was superfluous. Fixed with commit 1cbd45b824515109056989cf580ff447ad45d080.

**Comment by johanneskoester** [comment]
Ok, closing for now and looking forward to the separate issue.
